core/src/main/java/com/opensymphony/xwork2/ognl/OgnlUtil.java

```diff
@@ -74,6 +74,9 @@ public OgnlUtil() {
         excludedClasses = new HashSet<>();
         excludedPackageNamePatterns = new HashSet<>();
         excludedPackageNames = new HashSet<>();
+        excludedClasses = Collections.unmodifiableSet(excludedClasses);
```

Comment:
Hi @aleksandr-m .
If you're referring to `expressions ` being made final, then yes.  The IDE (NetBeans) pointed out it could be made final and I confirmed it is only referenced internally to the class, no getters/setters exist for it and no other code replaces the reference anywhere (always uses the initially constructed reference).
As for the exclude `Set` members any of them become unmodifiable, once their respective `setXXX `method is called at least once.  It follows then that they should always be unmodifiable when retrieved from a `getXXX `(even if a `setXXX `hasn't been called yet).